tests/integration/team_test.py

```diff
@@ -17,34 +17,106 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
 
+from contextlib import contextmanager
+import os
+import re
+
 import pytest
 
 import libnmstate
-from libnmstate.error import NmstateNotImplementedError
+from libnmstate.error import NmstateLibnmError
 from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
 from libnmstate.schema import InterfaceType
 from libnmstate.schema import Team
 
+from .testlib import assertlib
+from .testlib import cmd as libcmd
+
+
+RPM_NM_TEAM_PLUGIN_PATH = ("rpm", "-ql", "NetworkManager-team")
+
+SYSTEMCTL_RESTART_NM_CMD = ("systemctl", "restart", "NetworkManager")
 
 TEAM0 = "team0"
 
-TEAM0_STATE = {
-    Interface.KEY: [
-        {
-            Interface.NAME: TEAM0,
-            Interface.TYPE: InterfaceType.TEAM,
-            Team.CONFIG_SUBTREE: {
-                Team.PORT_SUBTREE: [{Team.Port.NAME: "eth1"}]
+
+@pytest.mark.skipif(
+    os.environ.get("CI") == "true",
+    reason="Team kmod not available in Travis CI",
+)
+def test_create_team_iface():
+    with team_interface(TEAM0) as team_state:
+        assertlib.assert_state(team_state)
+
+
+@pytest.mark.skipif(
+    os.environ.get("CI") == "true",
+    reason="Team kmod not available in Travis CI",
+)
+def test_edit_team_iface():
+    with team_interface(TEAM0) as team_state:
+        team_state[Team.CONFIG_SUBTREE] = {
+            Team.RUNNER_SUBTREE: {
+                Team.Runner.NAME: Team.Runner.RunnerMode.LOAD_BALANCE
             },
+            Team.PORT_SUBTREE: [{Team.Port.NAME: "eth1"}],
         }
-    ]
-}
+        libnmstate.apply(team_state)
 
 
-@pytest.mark.xfail(
-    raises=NmstateNotImplementedError,
-    reason="Team interface is not supported yet",
-    strict=True,
-)
-def test_sriov_not_implemented():
-    libnmstate.apply(TEAM0_STATE)
+def test_nm_team_plugin_missing():
+    with disable_nm_team_plugin():
+        with pytest.raises(NmstateLibnmError):
+            libnmstate.apply(
+                {
+                    Interface.KEY: [
+                        {
+                            Interface.NAME: TEAM0,
+                            Interface.TYPE: InterfaceType.TEAM,
+                            Interface.STATE: InterfaceState.UP,
+                        }
+                    ]
+                }
+            )
+
+
+@contextmanager
+def team_interface(ifname):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: ifname,
+                Interface.TYPE: InterfaceType.TEAM,
+                Interface.STATE: InterfaceState.UP,
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
+    try:
+        yield desired_state
+    finally:
+        team_dstate = desired_state[Interface.KEY][0]
+        team_dstate[Interface.STATE] = InterfaceState.ABSENT
+        libnmstate.apply(desired_state)
+
+
+@contextmanager
+def disable_nm_team_plugin():
```

Comment:
This context test is destructive, I would just either uninstall the rpm if possible or allow running it only in a container.

Reply:
This change was requested by @cathay4t, I prefer just to uninstall the rpm.. how can we check that we are in a container? And is @cathay4t okay with that?

Reply:
Gris' problem was that this breaks network connectivity and then he cannot re-install the RPM. Another workaround would be to run `dnf install --downloadonly NetworkManager-team` before breaking NM and installing the rpm again with `dnf install --cacheonly NetworkManager-team`.

Reply:
I like this workaround. Does this work for you @cathay4t? Thanks!

Reply:
`cacheonly` works for me.

Reply:
Done, we are now doing the `cacheonly` stuff. Thanks!